server/src/test/java/org/elasticsearch/search/aggregations/metrics/SumAggregatorTests.java

```diff
@@ -165,27 +193,193 @@ public void testSummationAccuracy() throws IOException {
+        );
+    }
+
+    public void testPartiallyUnmapped() throws IOException {
+        final MappedFieldType fieldType = new NumberFieldMapper.NumberFieldType(NumberType.LONG);
+        fieldType.setName(FIELD_NAME);
+        fieldType.setHasDocValues(true);
+
+        final SumAggregationBuilder builder = sum("_name")
+            .field(fieldType.name());
+
+        final int numDocs = randomIntBetween(10, 100);
+        final List<Set<IndexableField>> docs = new ArrayList<>(numDocs);
+        int sum = 0;
+        for (int i = 0; i < numDocs; i++) {
+            final long value = randomLongBetween(0, 1000);
+            sum += value;
+            docs.add(singleton(new NumericDocValuesField(fieldType.name(), value)));
+        }
+
+        try (Directory mappedDirectory = newDirectory(); Directory unmappedDirectory = newDirectory()) {
+            try (RandomIndexWriter mappedWriter = new RandomIndexWriter(random(), mappedDirectory)) {
+                mappedWriter.addDocuments(docs);
+            }
+
+            new RandomIndexWriter(random(), unmappedDirectory).close();
+
+            try (IndexReader mappedReader = DirectoryReader.open(mappedDirectory);
+                 IndexReader unmappedReader =  DirectoryReader.open(unmappedDirectory);
+                 MultiReader multiReader = new MultiReader(mappedReader, unmappedReader)) {
+
+                final IndexSearcher searcher = newSearcher(multiReader, true, true);
+
+                final InternalSum internalSum = search(searcher, newMatchAllQuery(), builder, fieldType);
+                assertEquals(sum, internalSum.getValue(), 0d);
+                assertTrue(AggregationInspectionHelper.hasValue(internalSum));
+            }
+        }
+    }
+
+    public void testValueScriptSingleValuedField() throws IOException {
+        scriptTestCase(1,
+            sum("_name")
+                .field(FIELD_NAME)
+                .script(new Script(ScriptType.INLINE, MockScriptEngine.NAME, VALUE_SCRIPT_NAME, emptyMap())));
+    }
+
+    public void testValueScriptMultiValuedField() throws IOException {
+        scriptTestCase(randomIntBetween(2, 5),
+            sum("_name")
+                .field(FIELD_NAME)
+                .script(new Script(ScriptType.INLINE, MockScriptEngine.NAME, VALUE_SCRIPT_NAME, emptyMap())));
+    }
+
+    public void testFieldScriptSingleValuedField() throws IOException {
+        scriptTestCase(1,
+            sum("_name")
+                .script(new Script(ScriptType.INLINE, MockScriptEngine.NAME, FIELD_SCRIPT_NAME, singletonMap("field", FIELD_NAME))));
+    }
+
+    public void testFieldScriptMultiValuedField() throws IOException {
+        scriptTestCase(randomIntBetween(2, 5),
+            sum("_name")
+                .script(new Script(ScriptType.INLINE, MockScriptEngine.NAME, FIELD_SCRIPT_NAME, singletonMap("field", FIELD_NAME))));
+    }
+
+    private void scriptTestCase(int numValuesPerField, SumAggregationBuilder builder) throws IOException {
```

Comment:
I don't *think* this name is right. It looks like it is just a generic "does random stuff reduce right" kind of thing.

Reply:
Had the same thought after writing it - the script specific part is the `final long expectedSum = sum + (numDocs * numValuesPerField);` because it expects the script to increment all the fields by 1

I'll see if I can make it less confusing

Reply:
I made it a little more general and moved some other test cases to it too. Still not thrilled with its readability